{{ cookiecutter.repo_name }}/src/data/make_dataset.py

```diff
@@ -1,30 +1,47 @@
-# -*- coding: utf-8 -*-
-import click
-import logging
+import os
+import io
+import requests
+import pandas as pd
 from pathlib import Path
-from dotenv import find_dotenv, load_dotenv
+from sklearn.preprocessing import StandardScaler
 
+DATA_LINK = 'https://archive.ics.uci.edu/ml/machine-learning-databases/blood-transfusion/transfusion.data'
+ROOT = Path(__file__).resolve().parents[2]
 
-@click.command()
-@click.argument('input_filepath', type=click.Path(exists=True))
-@click.argument('output_filepath', type=click.Path())
-def main(input_filepath, output_filepath):
-    """ Runs data processing scripts to turn raw data from (../raw) into
-        cleaned data ready to be analyzed (saved in ../processed).
+def massage_data(raw_data):
+    """ Preprocess the data for predictions
     """
-    logger = logging.getLogger(__name__)
-    logger.info('making final data set from raw data')
+    raw_data.rename(index=str, columns={"whether he/she donated blood in March 2007": "label"}, inplace = True)
+  
+    # generate features for year for time columns
+    for x,y in zip(['time_years','recency_years'],['Time (months)', 'Recency (months)']):
+        raw_data[x] = (raw_data[y]/12).astype('int')
+ 
+    # generate features for quarter for time columns (3 month periods)
+    for x,y in zip(['time_quarters','recency_quarters'],['Time (months)', 'Recency (months)']):
+        raw_data[x] = (raw_data[y]/3).astype('int')
 
+    return raw_data
+
+def dump_data(data, out_loc):
+    """ Given a path to a datafile, either a local file path
+    or a url, fetch the data and dump it to a csv
+    """
+    out_dir = os.path.join(ROOT, out_loc)
+    data.to_csv(out_dir, index=False)
 
-if __name__ == '__main__':
-    log_fmt = '%(asctime)s - %(name)s - %(levelname)s - %(message)s'
-    logging.basicConfig(level=logging.INFO, format=log_fmt)
 
-    # not used in this stub but often useful for finding various files
-    project_dir = Path(__file__).resolve().parents[2]
+def main():
+    """ Retrieves data and runs processing scripts to turn raw data from (../raw) into
+        cleaned data ready to be analyzed (saved in ../processed).
+    """
+    s=requests.get(DATA_LINK).content
```

Comment:
The downloading should happen with a `Makefile` rule (using `curl` or `wget`), using requests to `StringIO` works for tiny files but is a bad practice to show to learners.